### PR TITLE
ML-971: Add inner_join parameter to JoinWithTable.

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: integration/test_flow_integration.py [hunk 1]
```diff
@@ -256,19 +254,56 @@ def test_join_by_key_specific_attributes(setup_kv_teardown_test):
 
     controller = build_flow([
         SyncEmitSource(),
-        Filter(lambda x: x['col1'] > 8),
-        JoinWithTable(table, lambda x: x['col1'], attributes=['age']),
```
Comment (katyakats) [inline integration/test_flow_integration.py:260]: why did you delete the filter?

**Review by dinal — APPROVED** [review]

File: integration/test_flow_integration.py (hunk 1 above)
Comment (gtopper) [reply integration/test_flow_integration.py:260]: It seemed irrelevant to the test.
